glib-macros/src/boxed_derive.rs

```diff
@@ -161,6 +161,20 @@ pub fn impl_boxed(input: &syn::DeriveInput) -> TokenStream {
             }
         }
 
+        impl ::std::convert::From<#name> for #crate_ident::Value {
+            fn from(v: #name) -> Self {
+                unsafe {
+                    let ptr: *mut #name = ::std::boxed::Box::into_raw(::std::boxed::Box::new(v));
+                    let mut value = #crate_ident::Value::from_type(<#name as #crate_ident::StaticType>::static_type());
+                    #crate_ident::gobject_ffi::g_value_take_boxed(
+                        #crate_ident::translate::ToGlibPtrMut::to_glib_none_mut(&mut value).0,
```

Comment:
This needs to be `into_glib_ptr()` or otherwise you'll have a double-free

Reply:
Added an implementation of `IntoGlibPtr` to the macro